base/dict.jl

```diff
@@ -115,8 +115,44 @@ end
 in(k, v::KeyIterator) = !is(get(v.dict, k, secret_table_token),
                             secret_table_token)
 
+
+"""
+    keys(a::Associative)
+
+Return an iterator over all keys in a collection. `collect(keys(d))` returns an array of keys.
```

Comment:
worth noting here that the order shouldn't be relied upon as it depends on hash values?